libcst/_nodes/base.py

```diff
@@ -349,14 +362,12 @@ def deep_clone(self: _CSTNodeSelfT) -> _CSTNodeSelfT:
                 continue
             val = getattr(self, key)
 
-            # We can't use isinstance(val, CSTNode) here due to poor performance
-            # of isinstance checks against ABC direct subclasses. What we're trying
-            # to do here is recursively call this functionality on subclasses, but
-            # if the attribute isn't a CSTNode, fall back to copy.deepcopy.
-            try:
-                cloned_fields[key] = val.deep_clone()
-            except AttributeError:
-                cloned_fields[key] = deepcopy(val)
+            if isinstance(val, (str, bytes)):
+                cloned_fields[key] = _clone(val)
+            elif isinstance(val, Sequence):
+                cloned_fields[key] = tuple(_clone(v) for v in val)
+            else:
+                cloned_fields[key] = _clone(val)
```

Comment:
Isn't those isinstance check slow?

Can we do some micro benchmark to see how fast it is like using `%timeit` to clone a large tree?

Reply:
Experimentally, on a massive file on internal code in IG, this sped up cloning by a significant amount. However, I can take a similar approach where I attempt to iterate and if it fails I will fall back in the `except` case.